packages/svelte-virtual/src/index.ts

```diff
@@ -11,29 +11,53 @@ import {
 } from '@tanstack/virtual-core'
 export * from '@tanstack/virtual-core'
 
-import { readable, Readable } from 'svelte/store'
+import { derived, Readable, writable, Writable } from 'svelte/store'
+
+export type SvelteVirtualizer<
+  TScrollElement extends Element | Window,
+  TItemElement extends Element,
+> = Omit<Virtualizer<TScrollElement, TItemElement>, 'setOptions'> & {
+  setOptions: (
+    options: Partial<VirtualizerOptions<TScrollElement, TItemElement>>,
+  ) => void
+}
 
 function createVirtualizerBase<
   TScrollElement extends Element | Window,
   TItemElement extends Element,
 >(
   options: VirtualizerOptions<TScrollElement, TItemElement>,
-): Readable<Virtualizer<TScrollElement, TItemElement>> {
+): Readable<SvelteVirtualizer<TScrollElement, TItemElement>> {
+  let originalOnChange = options.onChange
+
   const virtualizer = new Virtualizer(options)
+  const originalSetOptions = virtualizer.setOptions
 
-  return readable(virtualizer, (set) => {
-    const cleanup = virtualizer._didMount()
-    virtualizer._willUpdate()
-    virtualizer.setOptions({
+  let virtualizerWritable: Writable<Virtualizer<TScrollElement, TItemElement>>
+
+  const setOptions = (
+    options: Partial<VirtualizerOptions<TScrollElement, TItemElement>>,
+  ) => {
+    originalOnChange = options.onChange || originalOnChange
+    originalSetOptions({
+      ...virtualizer.options,
       ...options,
       onChange: (instance: Virtualizer<TScrollElement, TItemElement>) => {
-        set(instance)
-        options.onChange?.(instance)
+        virtualizerWritable.set(instance)
+        originalOnChange?.(instance)
       },
     })
+    virtualizer._willUpdate()
+  }
```

Comment:
I think we can drop the `originalOnChange` and go with?
```ts
const setOptions = (
  options: Partial<VirtualizerOptions<TScrollElement, TItemElement>>,
) => {
  const resolvedOptions = {
    ...virtualizer.options,
    ...options,
  }
  originalSetOptions({
    ...resolvedOptions,
    onChange: (instance) => {
      virtualizerWritable.set(instance)
      resolvedOptions.onChange(instance)
    },
  })
  virtualizer._willUpdate()
}
  ```

Reply:
Cause the options type is `Partial<VirtualizerOptions<TScrollElement, TItemElement>>`, and then options maybe didn't have `onChange`, so it need call last `onChange`.

That's why we need `originalOnChange`.

Reply:
I think we don't need it, if merge options with virtualizer.options